redskyctl/internal/commands/export/export.go

```diff
@@ -0,0 +1,436 @@
+		Outputs: []kio.Writer{kio.ByteWriter{Writer: &experimentBuf}},
+	}
+	if err := experimentInput.Execute(); err != nil {
+		return err
+	}
+
+	// We don't want to bail if we cant find an experiment since we'll handle this later
+	if experimentBuf.Len() == 0 {
+		return nil
+	}
+
+	o.experiment = &redsky.Experiment{}
+
+	return commander.NewResourceReader().ReadInto(ioutil.NopCloser(&experimentBuf), o.experiment)
+}
+
+// filter returns a filter function to exctract a specified `kind` from the input.
+func filter(kind string) kio.FilterFunc {
+	return func(input []*yaml.RNode) ([]*yaml.RNode, error) {
+		var output kio.ResourceNodeSlice
+		for i := range input {
+			m, err := input[i].GetMeta()
+			if err != nil {
+				return nil, err
+			}
+			if m.Kind != kind {
+				continue
+			}
+			output = append(output, input[i])
+		}
+		return output, nil
+	}
+}
+
+func (o *Options) runner(ctx context.Context) error {
+	if o.trialName == "" {
+		return fmt.Errorf("a trial name must be specified")
+	}
+
+	if err := o.readInput(); err != nil {
+		return err
+	}
+
+	// See if we have been given an applcation
+	if err := o.extractApplication(); err != nil {
+		return fmt.Errorf("got an error when looking for application: %w", err)
+	}
+
+	// See if we have been given an experiment
+	if err := o.extractExperiment(); err != nil {
+		return fmt.Errorf("got an error when looking for experiment: %w", err)
+	}
+
+	switch {
+	case o.application != nil:
+		// Reset/Restrict application resources to only those specified by the application
+		// and resources generated by the generator
+		o.resources = make(map[string]struct{})
+		for _, file := range o.application.Resources {
+			o.resources[file] = struct{}{}
+		}
+
+		if err := o.generateExperiment(); err != nil {
+			return err
+		}
+	case o.experiment != nil:
+		// Dont need to do anything special
+	default:
+		return fmt.Errorf("unable to identify an experiment or application")
+	}
+
+	// At this point we must have an experiment
+	if o.experiment == nil {
+		return fmt.Errorf("unable to find an experiment")
+	}
+
+	// look up trial from api
+	trialItem, err := o.getTrialByID(ctx, o.experiment.Name)
+	if err != nil {
+		return err
+	}
+
+	trial := &redsky.Trial{}
+	experiment.PopulateTrialFromTemplate(o.experiment, trial)
+	server.ToClusterTrial(trial, &trialItem.TrialAssignments)
+
+	// render patches
+	var patches map[string]types.Patch
+	patches, err = createKustomizePatches(o.experiment.Spec.Patches, trial)
+	if err != nil {
+		return err
+	}
+
+	resourceNames := make([]string, 0, len(o.resources))
+	for name := range o.resources {
+		resourceNames = append(resourceNames, name)
+
+	}
+
+	yamls, err := kustomize.Yamls(
+		kustomize.WithFS(o.Fs),
+		kustomize.WithResourceNames(resourceNames),
+		kustomize.WithPatches(patches),
+	)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(o.Out, string(yamls))
+
+	return nil
+}
+
+func (o *Options) generateExperiment() error {
```

Comment:
add `app.filterByExperiment(o.trialName(minus id))` to reduce the application down to only the stuff we care about